spdx/parsers/tagvaluebuilders.py

```diff
@@ -1183,24 +1183,21 @@ def set_file_type(self, doc, type_value):
         Raise CardinalityError if more than one type set.
         Raise SPDXValueError if type is unknown.
         """
-        type_dict = {
-            "SOURCE": file.FileType.SOURCE,
-            "BINARY": file.FileType.BINARY,
-            "ARCHIVE": file.FileType.ARCHIVE,
-            "OTHER": file.FileType.OTHER,
-        }
         if not self.has_file(doc):
-            raise OrderError("File::Type")
+            raise OrderError("File::FileType")
 
-        if self.file_type_set:
-            raise CardinalityError("File::Type")
+        for file_type_enum in file.FileType:
+            if type_value == file_type_enum.name:
+                file_type = file_type_enum
+                break
```

Comment:
nit: Same as somewhere else, could be done more elegantly

Reply:
done